torch/csrc/api/include/torch/serialize.h

```diff
@@ -72,6 +73,8 @@ void save(const std::vector<torch::Tensor>& tensor_vec, SaveToArgs&&... args) {
   archive.save_to(std::forward<SaveToArgs>(args)...);
 }
 
+TORCH_API std::vector<char> save(const torch::IValue& ivalue);
```

Comment:
+1, this API doesn't seem to belong here - it returns `std::vector<char>`, while all other `torch::save` APIs don't return anything and expect to be used by `torch::save(value, filepath)`. I'd suggest naming this function `pickle_save` or similar, so that C++ API end users won't be confused.

btw, do we have this function in Python API? Or is it an implementation detail that would only be used by JIT?

Reply:
+1, this API doesn't seem to belong here - it returns `std::vector<char>`, while all other `torch::save` APIs don't return anything and expect to be used by `torch::save(value, filepath)`. I'd suggest naming this function `pickle_save` or similar, so that C++ API end users won't be confused.

btw, do we have this function in Python API? Or is it an implementation detail that would only be used by JIT?

Reply:
The function (renamed to `pickle_save`) should produce the same output as `torch.save()` in Python, so there should be a similar API here too. A rename is fine given the clash with the existing `torch::save`/`torch::load`. It's intended to be used as an API function e.g. #25591